### PR TITLE
Add support for opening the IDE from DevUI even if the default IDE script doesn't exist

### LINKED ISSUE #15721: Dev UI: open code in IDE should use IDE path from scanning processes
ATM we scan the processes to find which IDE to use, but then we discard the running process command, and invoke `idea` or `eclipse` hoping it's in the user's `$PATH`, which turns out is not necessarily true, for people with multiple versions of each IDEs.

It would be nice if the scanning of processes returned the exact IDE command path and we used that instead of relying on `$PATH`.

CC @geoand

### PR DESCRIPTION
**Pull request by geoand**
Currently support for this is only added for IntelliJ.
Code is opened client side (via the URL handler it installs)
so this support isn't needed, Netbeans already used something similar
and Eclipse seems to be sort of problematic because we don't have
a command which we can test that would not launch the IDE.

Furthermore this fallback is only tried if the default script does not exist

Fixes: #15721

### CODE REVIEW

**Comment by geoand** [comment]
I tested this on IntelliJ for both cases where `idea` exists on `$PATH` and where it doesn't

File: core/deployment/src/main/java/io/quarkus/deployment/ide/IdeUtil.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+final class IdeUtil {
+
+    private IdeUtil() {
+    }
+
+    static boolean isWindows() {
+        String os = System.getProperty("os.name");
+        return os.toLowerCase(Locale.ENGLISH).startsWith("windows");
+    }
+}
```
Comment (cuspymd) [inline core/deployment/src/main/java/io/quarkus/deployment/ide/IdeUtil.java:14]: `isWindows()` is currently used only in the "ide" package, but it seems not be dependent on only "ide". Isn't it better to change the class name to be more general? like 'SystemUtil' or 'PlatfromUtil'. And since the package '../deploy/util' already exists, wouldn't this class be better to be in it?

File: core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java [hunk 2]
```diff
@@ -1,23 +1,73 @@
...
 public enum Ide {
 
-    IDEA("idea"),
-    ECLIPSE("eclipse"),
-    VSCODE("code"),
-    NETBEANS("netbeans");
+    IDEA("idea", "--help"),
+    ECLIPSE("eclipse", (String[]) null),
+    VSCODE("code", "--version"),
+    NETBEANS(null);
```
Comment (phillip-kruger) [inline core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:15]: Why did you remove netbeans ? Running netbeans will still work if it is in your PATH. (In fact that is how mine works now, as the other way (finding the full path) has broken when I updated to netbeans 12.3, i still need to fix that.

**Review by phillip-kruger — APPROVED** [review]
I need to do some fixes on Netbeans anyway, so I can change that null back to netbeans.

File: core/deployment/src/main/java/io/quarkus/deployment/ide/IdeUtil.java (hunk 1 above)
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/IdeUtil.java:14]: Yes, if the need arises, I'll move it

File: core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java (hunk 2 above)
Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:15]: So simply running `netbeans ...` will work?
I was under the impression for the previous incarnation of the code that that could never happen for netbeans.

In order for me to fix it, I need to know if `netbeans --version` or `netbeans --help` will work without opening the IDE.
Can you please check and let me know?

Comment (phillip-kruger) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:15]: netbeans --help will give:

```
[pkruger@pkruger ~]$ netbeans --help
Usage: /opt/IDE/netbeans/platform/lib/nbexec {options} arguments

General options:
  --help                show this help 
  --jdkhome <path>      path to Java(TM) 2 SDK, Standard Edition
  -J<jvm_option>        pass <jvm_option> to JVM

  --cp:p <classpath>    prepend <classpath> to classpath
  --cp:a <classpath>    append <classpath> to classpath
Picked up _JAVA_OPTIONS: -Dawt.useSystemAAFontSettings=lcd
Module reload options:
  --reload /path/to/module.jar  install or reinstall a module JAR file

Additional module options:
  --modules                               
  --refresh                               Refresh all catalogs
  --list                                  Prints the list of all modules, their versions and enablement status
  --install <arg1>...<argN>               Installs provided JAR files as modules
  --disable <arg1>...<argN>               Disable modules for specified codebase names
  --enable <arg1>...<argN>                Enable modules for specified codebase names
  --update <arg1>...<argN>                Updates all or specified modules
  --update-all                            Updates all modules
  --extra-uc <arg>                        Add a extra Update Center (URL)
  --start-java-language-server [<arg>]    Starts the Java Language Server
  --start-java-debug-adapter-server <arg> Starts the Java Debug Adapter Server
  --open file1[:line1]...                 open specified file(s), possibly at given location; can also pass project directories
  --open-group NAME                       open a project group by shortened or full name (or unique substring)
  --close-group                           close any open project group
  --list-groups                           list available project groups

Core options:
  --laf <LaF classname> use given LookAndFeel class instead of the default
  --fontsize <size>     set the base font size of the user interface, in points
  --locale <language[:country[:variant]]> use specified locale
  --userdir <path>      use specified directory to store user settings
  --cachedir <path>     use specified directory to store user cache, must be different from userdir
  --nosplash            do not show the splash screen
```

if netbeans is in the PATH

Comment (phillip-kruger) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:15]: the other code can also be a fallback

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java:15]: Cool, I'll use that then

Comment (geoand) [reply core/deployment/src/main/java/io/quarkus/deployment/ide/Ide.java]: Fixed

**Review by phillip-kruger — APPROVED** [review]
